### DISCUSSION TITLE
HTTP parse error

### DISCUSSION THREAD

**Discussion by ideatells** [postalserver/postal#1721] [Help with using Postal]
Dear,

I just installed the postal server, but after the following command:
docker run -d \
   --name postal-caddy \
   --restart always \
   --network host \
   -v /opt/postal/config/Caddyfile:/etc/caddy/Caddyfile \
   -v /opt/postal/caddy-data:/data \
   caddy

when I go to https://<postal_IP>:5000
I am getting the err_ssl_protocol_error and in the log I see the following error message:

web_1       | 2021-11-22 08:12:28 +0000: HTTP parse error, malformed request (): #<Puma::HttpParserError: Invalid HTTP format, parsing fails.>

I searched around but I could not find the solution, can someone guide me to the solution?

**Comment by willpower232** [comment]
Port 5000 is the postal UI itself, it cannot handle https so `https://<postal_IP>:5000` will never work.

You should use `https://<your postal domain>` and not access port 5000 directly.

**Comment by ideatells** [comment]
Hi Willpower232,

THanks for your reply! When I go to https://<my postal domain> i am getting the same error:

ERR_SSL_PROTOCOL_ERROR

Do you have a solution for this?

**Reply by willpower232** [reply]
It means the SSL configuration in Caddy is not correct.

**Reply by ideatells** [reply]
How can I fix that, I followed the steps in the documentation and I got this error.

**Reply by willpower232** [reply]
I would try a server restart to be sure all your config changes are applying and if it still isn't working, can you share the caddy configuration you have?

**Reply by ideatells** [reply]
Hi willpower,

from the config file at location: /opt/postal/config/Caddyfile the following lines are stated in the file
 
post01xx.domain.com {
  reverse_proxy 127.0.0.1:5000

**Reply by willpower232** [reply]
Ah sorry I didn't realise the configuration file was so short for Caddy. I'm afraid I don't really know much about Caddy but if it hasn't generated your certificates properly then maybe you'll need to look in the Caddy logs
